docs/design/services/dynamodb/high-level-library/prototype/option-1/sync/Prototype.java

```diff
@@ -0,0 +1,1026 @@
+     *                                                                           .build());
+     *     }
+     * </code>
+     */
+    CreateGlobalSecondaryIndexResponse createGlobalSecondaryIndex(CreateGlobalSecondaryIndexRequest createRequest);
+
+    /**
+     * Invoke DynamoDB to create or override an {@link Item} in this table.
+     *
+     * This method is optimized for performance, and provides no additional response data. For additional options
+     * like conditions or consumed capacity, see {@link #putItem(PutItemRequest)}.
+     *
+     * Usage Example:
+     * <code>
+     *     try (DynamoDbDocumentClient client = DynamoDb.documentClient()) {
+     *         Table table = client.getTable("my-table");
+     *         table.putItem(Item.builder()
+     *                           .putAttribute("partition-key", UUID.randomUUID())
+     *                           .putAttribute("creation-time", Instant.now())
+     *                           .build());
+     *     }
+     * </code>
+     */
+    void putItem(Item item);
+
+    /**
+     * Invoke DynamoDB to create or override an {@link Item} in this table.
+     *
+     * This method provides more options than {@link #putItem(Item)}, like conditions or consumed capacity.
+     *
+     * Usage Example:
+     * <code>
+     *     try (DynamoDbDocumentClient client = DynamoDb.documentClient()) {
+     *         Table table = client.getTable("my-table");
+     *         table.putItem(PutItemRequest.builder()
+     *                                     .item(Item.builder()
+     *                                               .putAttribute("partition-key", "key")
+     *                                               .putAttribute("version", 2)
+     *                                               .build())
+     *                                     .condition("version = :expected_version")
+     *                                     .putConditionAttribute(":expected_version", 1)
+     *                                     .build());
+     *     } catch (ConditionFailedException e) {
+     *         System.out.println("Precondition failed.");
+     *         throw e;
+     *     }
+     * </code>
+     */
+    PutItemResponse putItem(PutItemRequest putRequest)
+            throws ConditionFailedException;
+
+    /**
+     * Invoke DynamoDB to create or override an Item in this table.
+     *
+     * This will convert the provided object into an {@link Item} automatically using the default Object-to-Item
+     * {@link ItemAttributeValueConverter}, unless an alternate converter has been overridden for the provided type.
+     *
+     * This method is optimized for performance, and provides no additional response data. For additional options
+     * like conditions or consumed capacity, see {@link #putObject(PutObjectRequest)}.
+     *
+     * Usage Example:
+     * <code>
+     *     public class MyItem {
```

Comment:
Can we infer without the attribute annotation?

Reply:
We could either have it be opt-in with @Attribute annotations or opt-out with @NotAttribute annotations. I don't have a strong opinion.